R/get_source_expressions.R

```diff
@@ -81,6 +82,31 @@ get_source_expressions <- function(filename) {
           double_quotes, capture(name = "starting", anything), double_quotes))
 
       line_location <- re_matches(source_file$content, rex(message_info$starting), locations = TRUE)
+      line_location <- line_location[complete.cases(line_location), ]
+
+      if (nrow(line_location) == 0L) {
+        if (grepl("attempt to use zero-length variable name", e$message)) {
+          # empty symbol: ``, ``(), ''(), ""(), fun(''=42), fun(""=42)
+          loc <- re_matches(source_file$content,
+            "``|''\\s*\\(|\"\"\\s*\\(|\\(''\\s*=|\\(\"\"\\s*=", locations = TRUE)
```

Comment:
I think we should stick to one type of regex specification in lintr.
Can you convert that to a `rex()` expression?

Reply:
NB the location of the argument in a function call need not be the first, eg. `fun(a = 1, '' = 2)`.
I think a regex covering this as well should be ``` ``|(''|"")\s*\(|[(,]\s*=(''|``|"")```

Reply:
Yes, you are right. Thanks!